src/java/org/apache/cassandra/locator/InetAddressAndPort.java

```diff
@@ -323,6 +331,52 @@ public static void initializeDefaultPort(int port)
         defaultPort = port;
     }
 
+    public static List<String> stringify(Iterable<InetAddressAndPort> endpoints)
+    {
+        return stringify(endpoints, true);
+    }
+
+    public static List<String> stringify(Iterable<InetAddressAndPort> endpoints, boolean withPort)
+    {
+        List<String> stringEndpoints = new ArrayList<>();
+        for (InetAddressAndPort ep : endpoints)
+        {
+            stringEndpoints.add(ep.getHostAddress(withPort));
+        }
+        return stringEndpoints;
+    }
+
+    /**
+     * Parses a comma-separated list of hosts to a set of {@link InetAddressAndPort}
+     *
+     * @param value       the comma-separated list of hosts to parse
+     * @param failOnError whether to fail when encountering an invalid hostname
+     * @return the set of parsed {@link InetAddressAndPort}
+     */
+    public static Set<InetAddressAndPort> parseHosts(String value, boolean failOnError)
+    {
+        Set<InetAddressAndPort> hosts = new HashSet<>();
+        for (String host : Splitter.on(',').split(value))
+        {
+            try
+            {
+                hosts.add(InetAddressAndPort.getByName(host));
+            }
+            catch (UnknownHostException e)
+            {
+                if (failOnError)
+                {
+                    throw new IllegalArgumentException(e.getMessage());
```

Comment:
nit: 
throw new IllegalArgumentException("Failed to parse host: " + host, e);

Reply:
I will change the message. Unfortunately we cannot propagate the exception because it might break JMX when the exception is not a java type.

Reply:
I will change the message. Unfortunately we cannot propagate the exception because it might break JMX when the exception is not a java type.

Reply:
I guess for the use cases we have, we don't throw. So I think it's fine to have it as you suggest.